drivers/serial/uart_ns16550.c

```diff
@@ -657,6 +664,16 @@ static void uart_ns16550_irq_tx_enable(const struct device *dev)
 {
 	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
 
+#ifdef CONFIG_PM
+	struct uart_ns16550_dev_data *const dev_data = dev->data;
+
+	for (int i = ARRAY_SIZE(pm_states) - 1; i >= 0; i--) {
+		if (!dev_data->tx_stream_on) {
+			dev_data->tx_stream_on = true;
+			pm_constraint_set(pm_states[i].state);
```

Comment:
this loop will unnecessarily keep iterating after if evaluates to true. I'm failing to fully understand the combination of the for and this if inside. Can this be explained better?

Reply:
The loop only needs to do constraint set according to different power states. Fixed.